File: src/SnapToTable.Domain/Entities/Recipe.cs
namespace SnapToTable.Domain.Entities;

public record Recipe(
    Guid RecipeAnalysisId,
    string Name,
    string Category,
    string Url,
    TimeSpan? PrepTime,
    TimeSpan? CookTime,
    TimeSpan? AdditionalTime,
    int? Servings,
    IReadOnlyList<string> Ingredients,
    IReadOnlyList<string> Directions,
    IReadOnlyList<string> Notes) : BaseEntity;

File: client/snap-to-table/features/recipes/types/recipe.ts
export interface Recipe {
    id: string;
    createdAt: Date;
    recipeAnalysisId: string;
    name: string;
    category: string;
    prepTimeInMinutes: number | null;
    cookTimeInMinutes: number | null;
    additionalTimeInMinutes: number | null;
    totalTimeInMinutes: number | null;
    servings: number | null;
    ingredients: readonly string[];
    directions: readonly string[];
    notes: readonly string[];
}